File: backends/php-laravel/app/Domain/Staff/Entities/Staff.php
<?php
/**
 * This is a program developed by BobTabo.
 *
 * Copyright (c) 2026 BobTabo. All Rights Reserved.
 */
namespace App\Domain\Staff\Entities;

use App\Domain\Staff\Enums\Provider;
use App\Domain\Staff\Enums\StaffRole;
use App\Domain\Staff\Enums\StaffStatus;
use App\Support\Entities\AbstractEntity;
use Carbon\Carbon;

/**
 * スタッフEntityクラスです。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 * @package App\Domain\Staff\Entities
 */
class Staff extends AbstractEntity
{
    public ?int $id = null;
    public ?string $name = null;
    public ?string $email = null;
    public ?Provider $provider = null;
    public ?string $providerId = null;
    public ?string $avatar = null;
    public ?StaffRole $role = null;
    public ?StaffStatus $status = null;
    public ?Carbon $lastLoginAt = null;
}

File: backends/java-spring/src/main/java/com/authorization/domain/staff/entities/Staff.java
/**
 * This is a program developed by BobTabo.
 *
 * Copyright (c) 2026 BobTabo. All Rights Reserved.
 */
package com.authorization.domain.staff.entities;

import com.authorization.domain.staff.enums.Provider;
import com.authorization.domain.staff.enums.StaffRole;
import com.authorization.domain.staff.enums.StaffStatus;
import com.authorization.support.entities.AbstractEntity;
import java.time.LocalDateTime;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * スタッフEntityクラスです。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
@Data
@EqualsAndHashCode(callSuper = true)
public class Staff extends AbstractEntity {

    private Long id;
    private String name;
    private String email;
    private Provider provider;
    private String providerId;
    private String avatar;
    private StaffRole role;
    private StaffStatus status;
    private LocalDateTime lastLoginAt;
}
